### PR TITLE
Test Transform Script

### LINKED ISSUE #20: Create unit tests for transform script
# Ticket

## Description:
- Develop unit tests for the transform script to ensure raw data is cleaned and normalised correctly.

## Outcomes:
- Tests validate data cleaning functions
- Coverage for edge cases like missing data

## User Stories:
- As a developer, I want unit tests for the transform script so that data processing is reliable and consistent.

### PR DESCRIPTION
**Pull request by jacoblewis046**
closes #20

### CODE REVIEW

File: bluesky_etl/test_transform.py [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+    @patch('transform.pipeline')
+    def test_transform_multiple_topics(self, mock_pipeline, transformer):
+        """Test transformation with multiple topics found"""
+        mock_pipeline_instance = Mock()
+        mock_pipeline.return_value = mock_pipeline_instance
+        mock_pipeline_instance.return_value = [
+            {'label': 'POS', 'score': 0.8}
+        ]
+
+        # message with multiple topics
```
Comment (KirstineInvierno) [inline bluesky_etl/test_transform.py:289]: same with this, I think we can remove this comment

**Review by KirstineInvierno — CHANGES_REQUESTED** [review]
Really thorough test suit! Just left some minor nitpicks.

File: bluesky_etl/load.py [hunk 2]
```diff
@@ -49,7 +49,12 @@ def upload_df_to_mention(self, df: pd.DataFrame,
...
+    time1 = time.time()
     print('connecting')
     con = loader.get_sql_conn()
-    print('connected')
+    time2 = time.time()
+    print(f'connected in {round(time2-time1, 2)} seconds')
+    print('uploading')
     loader.upload_df_to_mention(engine=con, df=df, schema='bluesky')
+    time3 = time.time()
+    print(f'uploaded in {round(time3-time2, 2)} seconds')
```
Comment (KirstineInvierno) [inline bluesky_etl/load.py:60]: Question: Do we need lines 52, 53, 54-57, 59-60? Or were they for when we were trying to work out how long it took to run?

File: bluesky_etl/transform.py [hunk 3]
```diff
@@ -1,107 +1,218 @@
...
+            f"transform script complete in {round(time2-time1, 2)} seconds")
+
+        return pd.concat(dataframes, ignore_index=True)
 
 
 def main():
-    """main function"""
-    print(transform(None))
+    """Main function"""
+    message = Message({
```
Comment (KirstineInvierno) [inline bluesky_etl/transform.py:197]: question: also do we need these?

File: bluesky_etl/load.py (hunk 2 above)
Comment (jacoblewis046) [reply bluesky_etl/load.py:60]: I still think they're good to have, I should maybe make them logging rather than print statements though

File: bluesky_etl/transform.py (hunk 3 above)
Comment (jacoblewis046) [reply bluesky_etl/transform.py:197]: That's just for pylint
